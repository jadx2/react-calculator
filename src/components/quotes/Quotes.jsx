import quotes from './quotes';
import '../styles/quotes.css';

const Quotes = () => {
  const randomQuote = quotes[Math.floor(Math.random() * quotes.length)];
  return (
    <div className="quotes">
      <h2 className="quotes__quote">{randomQuote.quote}</h2>
      <p className="quotes__author">
        -
        {randomQuote.author}
      </p>
    </div>
  );
};

export default Quotes;
