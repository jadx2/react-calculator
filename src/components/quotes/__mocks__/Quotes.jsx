const mockedQuotes = () => <div>This is the quotes page!</div>;

export default mockedQuotes;
