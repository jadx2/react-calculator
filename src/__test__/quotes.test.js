import TestRenderer from 'react-test-renderer';
import Quotes from '../components/quotes/Quotes';

jest.mock('../quotesData/quotes');
it('renders Quotes', () => {
  const tree = TestRenderer.create(<Quotes />).toJSON();
  expect(tree).toMatchSnapshot();
});
