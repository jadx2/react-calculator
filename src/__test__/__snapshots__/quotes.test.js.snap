// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Quotes 1`] = `
<div
  className="quotes"
>
  <h2
    className="quotes__quote"
  >
    Mathematics is a game played according to certain rules with meaningless marks on paper.
  </h2>
  <p
    className="quotes__author"
  >
    -
    David Hilbert
  </p>
</div>
`;
